zhook.py

```diff
@@ -344,70 +342,172 @@ def addWatchDetails(self):
-
-  if zitiId is None:
-    print("ERROR: no Ziti identity provided, set INPUT_ZITIID or INPUT_ZITIJWT")
-    exit(1)
+    # Support inline JSON or base64-encoded identity JSON from a single variable
+    zitiIdInput = os.getenv("INPUT_ZITIID")
+
+    # Prefer valid inline JSON if present
+    if zitiIdInput and _try_parse_json(zitiIdInput):
+      zitiIdJson = zitiIdInput
+      zitiIdContext = json.loads(zitiIdInput)
+      print("Detected valid inline JSON in INPUT_ZITIID")
+    else:
+      # Try decoding inline as base64 if provided and not valid JSON
+      decodedInline = _try_decode_b64_to_json_str(zitiIdInput) if zitiIdInput else None
+      if decodedInline is not None:
+        zitiIdEncoding = zitiIdInput
+        zitiIdJson = decodedInline
+        zitiIdContext = json.loads(decodedInline)
+        print("Detected base64-encoded identity in INPUT_ZITIID and decoded it")
+
+    if zitiIdJson is None:
+      print("ERROR: no Ziti identity provided, set INPUT_ZITIID (inline JSON or base64-encoded), or INPUT_ZITIJWT")
+      exit(1)
+
+  # Keep a small helper for safe string hints (used only in error/debug prints if needed)
+  def _safe_hint(s):
+    if s is None:
+      return "<none>"
+    hint_len = len(s)
+    head = s[:8].replace('\n', ' ')
+    return f"len={hint_len}, startswith='{head}...'"
 
   idFilename = "id.json"
   with open(idFilename, 'w') as f:
-    f.write(zitiId)
-    openziti.load(idFilename)
+    f.write(zitiIdJson)
+
+  # Defer openziti.load() until inside the monkeypatch context to keep
+  # initialization/teardown paired and avoid double-free on shutdown.
 
   # Create webhook body
   try:
-    mwb = MattermostWebhookBody(username, icon, channel, eventName, eventJsonStr, actionRepo)
+    mwb = MattermostWebhookBody(username, icon, eventName, eventJson, actionRepo)
   except Exception as e:
     print(f"Exception creating webhook body: {e}")
     raise e
 
   # Post the webhook over Ziti
-  headers = {'Content-Type': 'application/json'}
-  data = mwb.dumpJson()
-  debug = os.getenv("ZHOOK_DEBUG", "").casefold() == "true"
+  # Build dict payload; requests will set Content-Type when using json=
+  payload = mwb.body
 
-  try:
-      with openziti.monkeypatch():
-          if debug:
-              print(f"Posting webhook to {url} with headers {headers} and data {data}")
-          else:
-              print(f"Posting webhook to {url} with headers {headers}")
-
-          r = requests.post(url, headers=headers, data=data)
-          print(f"Response Status: {r.status_code}")
-
-          if debug:
-              print(f"Response HEADERS: {r.headers}")
-              print(f"Response CONTENT: {r.content}")
-
-          if 200 <= r.status_code < 300:
-              print(f"INFO: successfully posted. status code {r.status_code}")
-              sys.exit(0)
-          else:
-              print(f"ERROR: unexpected status code {r.status_code}")
-              sys.exit(1)
+  with openziti.monkeypatch():
```

Comment:
the monkeypatch covers too much. it would be a little clearer
if you extracted a function to do the request work:
```
@openziti.zitify()
def doPost() -> requests.Response:
    # does it need a session? it is a single request
    session = requests.Session()
    return session.post(...)
```